offline/cable_cru_TRENDY.F90

```diff
@@ -365,104 +380,412 @@ SUBROUTINE CRU_GET_SUBDIURNAL_MET(CRU, MET, CurYear, ktau, kend)
+  !*## Purpose
+  !
+  ! Get the day and year to be used for the meteorology input.
+  !
+  !## Method
+  !
+  ! Use the SimYear and the recycling parameters to determine the date to use
+  ! for the meteorology. The SimDay is checked for validity, as it may be called
+  ! with SimDay+-N at the calling site. If the meteorology is recycled, then
+  ! the year is determined by:
+  ! ```
+  ! RecycleStart + MODULO(SimYear - 1501, RecyclePriod)
+  ! ```
+
+  INTEGER, INTENT(IN) :: SimYear, SimDay, RecycleStart, RecyclePeriod
+  LOGICAL, INTENT(IN) :: IsRecycled, LeapYears
+
+  INTEGER, INTENT(OUT) :: MetYear, MetDay
+
+  ! On leap years and the next/previous day values, and the tricky handling at
+  ! the start and end of recycling periods. The approach for now is to only
+  ! check for recycling at the start of the routine, before doing any year
+  ! shifts when the day = 0 or DaysInYear. This means it is possible to retrieve
+  ! meteorology from outside the recycling period when handling neighbouring
+  ! days.
+  ! For example, we have a recycling period from 1951 to 1960, and we want the
+  ! meteorology from 01/01/1951, and consequently some of the previos day's
+  ! meteorology. The "previous day" with respect to the meteorology can be
+  ! either 31/12/1950 or 31/12/1960, depending on when the MODULO is applied.
+  ! The current approach would use 31/12/1949 as the previous day, which is less
+  ! likely to cause issues with large jumps in the time-local conditions.
+  ! Also simplifies leap year handling. In that instance, we can simply check
+  ! whether the new year is a leap year or not, instead of passing the new year
+  ! through the MODULO calculation again.
+
+  ! We need to know how many days we expect in the year so check for end of year
+  ! behaviour
+  INTEGER :: DaysInYear = 365
+
+  ! Set up the first pass at recycling
+  MetYear = SimYear
+  IF (IsRecycled) THEN
+    MetYear = RecycleStart + MODULO(MetYear - 1501, RecyclePeriod)
   END IF
 
-  ! Was the Tmin recycled?
-  IF (CRU%isRecycled(Tmin)) THEN
-    DummyYear = CRU%RecycStartYear + MOD(DummyYear - 1501, CRU%metRecyc)
+  ! How many days in the year, for correct end of year handling
+  IF (LeapYears) THEN
+    DaysInYear = DaysInYear + leap_day(MetYear)
   END IF
 
-  CALL read_metvals(CRU%MetDatasets(Tmin), CRU%Met(nextTmin)%MetVals, land_x,&
-    land_y, DummyYear, DummyDay, CRU%LeapYears, CRU%xDimSize, CRU%yDimSize,&
-    CRU%DirectRead)
+  ! Check the validity of the passed day
+  IF (SimDay <= 0) THEN
+    ! Go back to last year- set the day later, once we know whether the MetYear
+    ! is a leapyear or not
+    MetYear = SimYear - 1
 
-END SUBROUTINE cru_get_daily_met
+    ! This handles any future scenarios where we may change the day by >1
+    MetDay = 365 + SimDay
+    IF ((LeapYears) .AND. is_leapyear(MetYear)) THEN
+      MetDay = 366 + SimDay
+    END IF
+  ELSEIF (SimDay == (DaysInYear + 1)) THEN
+    MetYear = SimYear + 1
+    MetDay = SimDay - DaysInYear
```

Comment:
Yep, should definitely by ```>=``` rather than ```==```. I originally built it specifically for +- one day, but figured why not make it a bit more generic?